url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_FromClauseParser.java

```diff
@@ -0,0 +1,6078 @@
+
+							}
+							break;
+
+						default :
+							break loop38;
+						}
+					}
+
+					}
+					break;
+
+			}
+
+			RPAREN115=(Token)match(input,RPAREN,FOLLOW_RPAREN_in_partitionedTableFunction1624); if (state.failed) return retval; 
+			if ( state.backtracking==0 ) stream_RPAREN.add(RPAREN115);
+
+			// FromClauseParser.g:237:16: (alias= Identifier )?
+			int alt40=2;
+			alt40 = dfa40.predict(input);
+			switch (alt40) {
+				case 1 :
+					// FromClauseParser.g:237:16: alias= Identifier
+					{
+					alias=(Token)match(input,Identifier,FOLLOW_Identifier_in_partitionedTableFunction1628); if (state.failed) return retval; 
+					if ( state.backtracking==0 ) stream_Identifier.add(alias);
+
+					}
+					break;
+
+			}
+
+			// AST REWRITE
+			// elements: alias, expression, name, partitioningSpec, partitionTableFunctionSource
+			// token labels: alias, name
+			// rule labels: retval
+			// token list labels: 
+			// rule list labels: 
+			// wildcard labels: 
+			if ( state.backtracking==0 ) {
+			retval.tree = root_0;
+			RewriteRuleTokenStream stream_alias=new RewriteRuleTokenStream(adaptor,"token alias",alias);
+			RewriteRuleTokenStream stream_name=new RewriteRuleTokenStream(adaptor,"token name",name);
+			RewriteRuleSubtreeStream stream_retval=new RewriteRuleSubtreeStream(adaptor,"rule retval",retval!=null?retval.getTree():null);
+
+			root_0 = (CommonTree)adaptor.nil();
+			// 238:4: -> ^( TOK_PTBLFUNCTION $name ( $alias)? partitionTableFunctionSource ( partitioningSpec )? ( expression )* )
+			{
+				// FromClauseParser.g:238:9: ^( TOK_PTBLFUNCTION $name ( $alias)? partitionTableFunctionSource ( partitioningSpec )? ( expression )* )
+				{
+				CommonTree root_1 = (CommonTree)adaptor.nil();
+				root_1 = (CommonTree)adaptor.becomeRoot((CommonTree)adaptor.create(TOK_PTBLFUNCTION, "TOK_PTBLFUNCTION"), root_1);
+				adaptor.addChild(root_1, stream_name.nextNode());
+				// FromClauseParser.g:238:35: ( $alias)?
+				if ( stream_alias.hasNext() ) {
+					adaptor.addChild(root_1, stream_alias.nextNode());
+				}
+				stream_alias.reset();
+
+				adaptor.addChild(root_1, stream_partitionTableFunctionSource.nextTree());
+				// FromClauseParser.g:238:71: ( partitioningSpec )?
+				if ( stream_partitioningSpec.hasNext() ) {
+					adaptor.addChild(root_1, stream_partitioningSpec.nextTree());
+				}
+				stream_partitioningSpec.reset();
+
+				// FromClauseParser.g:238:89: ( expression )*
+				while ( stream_expression.hasNext() ) {
+					adaptor.addChild(root_1, stream_expression.nextTree());
+				}
+				stream_expression.reset();
+
+				adaptor.addChild(root_0, root_1);
+				}
+
+			}
+
+
+			retval.tree = root_0;
+			}
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			if ( state.backtracking==0 ) {
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+			}
+			if ( state.backtracking==0 ) { gParent.msgs.pop(); }
+		}
+
+		catch (RecognitionException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)